build.sbt

```diff
@@ -732,7 +732,7 @@ def commonScalacOptions(scalaVersion: String) =
     "-language:implicitConversions",
     "-language:experimental.macros",
     "-unchecked",
-    "-Ywarn-dead-code",
+    //"-Ywarn-dead-code",
```

Comment:
This probably needs to be discussed, it's required to define `empty`. We don't, in this expression of the type class, end up using `empty` much, so it might make more sense to drop `empty` than this, but I'd like to hear opinions before I go for it.

Reply:
When you say `empty` you mean `zero`? :)

Reply:
I think we should go with `def zero[A]: F[A]` then to make it easier. `Nothing` can be weird in a lot of ways :smile: